clients/segments/segments.go

```diff
@@ -148,16 +137,16 @@ func (c Client) Upsert(ctx context.Context, id string, data []byte) (Response, e
 
 	err = json.Unmarshal(existingResourceBody, &getResponse)
 	if err != nil {
-		return Response{}, fmt.Errorf(errMsgWithId, "upsert", id, err)
+		return Response{}, fmt.Errorf(errMsgWithId, "update", id, err)
 	}
 	if getResponse.Version == 0 {
 		return Response{}, fmt.Errorf("missing version field in API response")
 	}
 
-	//Adds uid and owner if not set
+	//Adds uid and owner if not set(they are mandatory from the API)
 	data, err = addOwnerAndUIDIfNotSet(data, getResponse.Owner, id)
```

Comment:
Is it useful to let a user of this method have the ability to set the owner and version manually?

If we expect the user to handle version and owner, it's cheaper to see if it is already set. And only if it isn't go to the server and get that information (it's more important to optimize the network than cpu consumption)

Reply:
The handling is for only `owner` and `uid`. So in any case, we need to make a get for the `version`. So no call can be saved there.